### PR TITLE
[FIX] Backwards compatibility CollectiblesController

### PR DESCRIPTION
**Pull request by gantunesr**
# Description

With version 18.0.0 of the controllers, there was on the data type of the property `tokenId` (`Number -> String`), this update ensures that the app works properly with both data types.

## ⚠️  Blockers 

- PR #3380 

# Tasks

- [x] Remove previous fix.
- [x] Add controllers update
- [ ] Update snapshots (if necessary) 

# References

- [MM Controllers PR](https://github.com/MetaMask/controllers/pull/630)

### CODE REVIEW

File: app/core/Engine.js [hunk 1]
```diff
@@ -173,18 +174,22 @@ class Engine {
...
+					addTokens: tokensController.addTokens.bind(tokensController),
+					getTokensState: () => tokensController.state,
+					getTokenListState: () => tokenListController.state,
+				}),
+				new CollectibleDetectionController({
+					onCollectiblesStateChange: (listener) => collectiblesController.subscribe(listener),
+					onPreferencesStateChange: (listener) => preferencesController.subscribe(listener),
+					onNetworkStateChange: (listener) => networkController.subscribe(listener),
 					getOpenSeaApiKey: () => collectiblesController.openSeaApiKey,
 					getBalancesInSingleCall:
```
Comment (Cal-L) [inline app/core/Engine.js:190]: I believe `getBalancesInSingleCall` belongs under `TokenDetectionController`

**Review by Cal-L — CHANGES_REQUESTED** [review]

File: app/core/Engine.js [hunk 2]
```diff
@@ -597,7 +603,6 @@ export default {
 			AddressBookController,
 			AssetsContractController,
 			CollectiblesController,
-			AssetsDetectionController,
```
Comment (Cal-L) [inline app/core/Engine.js:600]: I'm pretty sure both `TokenDetectionController` and `CollectibleDetectionController` is needed here for the engine  reducer to recognize the state if these controllers. Here's how the reducer is accessing the Engine.state: `newState.backgroundState[action.key] = Engine.state[action.key];`

Comment (gantunesr) [reply app/core/Engine.js:600]: You're right! This change is from this PR https://github.com/MetaMask/metamask-mobile/pull/3380 which is a blocker to this one

File: app/core/Engine.js (hunk 1 above)
Comment (gantunesr) [reply app/core/Engine.js:190]: Thats true! This should be update on PR #3380 which is blocking this one

Comment (gantunesr) [reply app/core/Engine.js:190]: Thats true! This should be update on PR #3380 which is blocking this one

File: app/core/Engine.js (hunk 2 above)
Comment (gantunesr) [reply app/core/Engine.js:600]: You're right, this should be fixed in #3380 which is blocking this PR

**Comment by gantunesr** [comment]
This PR is no longer necessary

**Pull request closed without merging**